crates/gitbutler-diff/src/diff.rs

```diff
@@ -96,17 +96,17 @@ impl GitHunk {
 
 /// Comparison
 impl GitHunk {
-    /// integration_intersects_unapplied is used to determine if a hunk from a diff between integration and the trunk intersects with an unapplied hunk.
+    /// workspace_intersects_unapplied is used to determine if a hunk from a diff between workspace and the trunk intersects with an unapplied hunk.
     /// We want to use the new start/end for the integraiton hunk and the old start/end for the unapplied hunk.
-    pub fn integration_intersects_unapplied(
-        integration_hunk: &GitHunk,
+    pub fn workspace_intersects_unapplied(
```

Comment:
The naming here feels a little janky. (I know I wrote this originally).

What do we want to call something that is localized to a branch.

Workspace-scoped or just "workspace_x" feels reasonable for things in the combined scope, but what do we want for things that are branch-scoped?

Also, you missed a miss-spelling of integration in the comment 😅

Reply:
Thanks for spotting spelling mistakes!

About the name, by unapplied_hunk, do we really mean uncommitted_hunk? I also see "unstaged_hunk", but that to me feels like a misnomer since we do not engage in staging of changes.

Reply:
Hmmmm...

Taking a look at the usage.

`integration_hunk` is hunk which has been created by diffing the head of an individual virtual branch against the base commit. (of note, its not the integration commit that is getting diffed, which may cause line number differences if the same file has had additions in two different branches)
`unapplied_hunk` is an uncommitted hunk that has been created by diffing all uncommitted changes against the integration commit.


renaming `unapplied_hunk` to `uncommitted_hunk` makes sense. I think `integration_hunk` might best be renamed to `branch_hunk`

Reply:
I had a look, a few things need renaming in the surrounding code. Let's land this and follow-up with a separate pr for that?